src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/FunctionProbes/EventSourceEmittingProbes.cs

```diff
@@ -29,18 +33,20 @@ public void CacheMethods(IList<MethodInfo> methods)
 
         public bool EnterProbe(ulong uniquifier, object[] args)
         {
-            // We allow the instrumentation of system types, but these types can also be part of an ILogger implementation.
-            // In addition, certain loggers don't log directly, but into a background thread.
-            // We guard against reentrancy on the same thread.
-            // All types that start with System & Microsoft do not invoke ILogger directly. Rather, they queue a message to a dedicated background thread.
-            // All probes that are triggered from the dedicated background thread do not log.
-            // All probes from the known console logger processor thread do not log.
-
-            // Possible additional guards in the future:
-            // If we instrument Modules instead of specific methods, we may need to exclude certain types to prevent noise. (such as System.String)
-            // If other custom loggers create background threads, we may need a way to specify exclusions for those threads.
+            try
+            {
+                return EnterProbeInternal(uniquifier, args);
+            }
+            catch (Exception ex)
```

Comment:
This changes the probe faulting behavior -- the current behavior is that if a probe faults for whatever reason, all probes will automatically get uninstalled. Let's restore that original behavior (by not swallowing the exception). If we want to change the faulting behavior I'd prefer that to be a separate PR with more discussion around it.

As for keeping this try/catch and reporting back the exception, what does this PR do with the `FailedToCapture` event?

Reply:
It was an artifact of development, it shouldn't have made it in the PR. I reverted the change